src/NHibernate/Dialect/Dialect.cs

```diff
@@ -2393,5 +2394,12 @@ public virtual ISQLExceptionConverter BuildSQLExceptionConverter()
 			// may override to return whatever is most appropriate for that vendor.
 			return new SQLStateConverter(ViolatedConstraintNameExtracter);
 		}
+
+		internal static string GetDriverName<TFallback>(string preferredName) where TFallback : IDriver
+		{
+			return ReflectHelper.ClassForFullNameOrNull(preferredName) == null
+				? typeof(TFallback).FullName
+				: preferredName;
```

Comment:
Checking this again, I realize it generates a lot of exceptions when it needs to fallback, because it is called at each dialect creation, and dialects are created numerous times even for a single test. Even when it does not need to fallback, being called frequently is not a good thing since it performs reflection.

So I propose to cache the name checking:

```c#
private static readonly ConcurrentDictionary<string, System.Type> DriverClasses = new ConcurrentDictionary<string, System.Type>();

internal static string GetDriverName<TFallback>(string preferredName) where TFallback : IDriver
{
	return DriverClasses.GetOrAdd(preferredName, ReflectHelper.ClassForFullNameOrNull) == null
		? typeof(TFallback).FullName
		: preferredName;
}
```